include/zephyr/bluetooth/hci_types.h

```diff
@@ -3216,6 +3216,152 @@ struct bt_hci_evt_le_biginfo_adv_report {
 #define BT_HCI_ERR_TOO_LATE                     0x46
 #define BT_HCI_ERR_TOO_EARLY                    0x47
 
+static inline const char *bt_hci_err_to_str(uint8_t err)
```

Comment:
Is the toolchain able to de-duplicate this function? If it is instantiated once for every compilation unit it's used in and doesn't optimize, then we should move the function definition to a implementation file.

Reply:
Started out by creating https://github.com/zephyrproject-rtos/zephyr/pull/73826